dist/comment_style.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+import argparse, os, re, subprocess, sys
+
+def print_msg(file_name, line_num, line, multiline):
+    print("Illegal " + multiline + "comment in " + file_name + ":" + str(line_num - 1) + " " +
+        line.strip('\n'))
+
+def check_c_comments(file_name):
+    count = 0
+    with open(file_name) as f:
+        line_num = 0
+        for line in f:
+            stripped = line.strip()
+            if match(stripped):
+                print_msg(file_name, line_num, line, "")
+                count +=1
+            line_num += 1
+    return count
+
+def match(stripped_line):
+    # We don't worry about whitespace in matches as we strip the line of whitespace.
+    url = re.compile(r'https?:\/\/')
+
+    # // Style comments
+    single_line_comment = re.compile(
```

Comment:
A full explanation is gonna be tough, but we can we some details as to what cases we're catching here?
Something like:
```
Detect all occurences of `//` on a line. This can be at the start of a line, or after preceding text
```

I'm not sure exactly what case we're handling in this line so I'll leave that to you

Reply:
Also as we're not making use of captured groups we can drop the `?:`s here.

Reply:
I reviewed the regex and realised a part of it was redundant. I've added a comment in bf9d525 that also addresses your `?:` comment.